dsc-linux-5.4.173/patches-2023-01/01_elba_kpcimgr.patch

```diff
@@ -0,0 +1,1463 @@
+From 19bc58d0416d79ad3dc66d5be3bb641be86576a5 Mon Sep 17 00:00:00 2001
+From: Hiren Mehta <hiren@pensando.io>
+Date: Fri, 13 Jan 2023 16:41:44 -0800
+Subject: [PATCH] drivers/soc/pensando: kpciemgr driver.
+
+kpciemgr driver enables the relocation of the module code to handle
+Elba indirect PCIe transactions.
+
+Signed-off-by: Hiren Mehta <hiren@pensando.io>
+---
+ arch/arm64/boot/dts/pensando/elba.dtsi |  11 +
+ drivers/soc/pensando/Makefile          |   2 +
+ drivers/soc/pensando/kpci_constants.h  |  41 ++
+ drivers/soc/pensando/kpcimgr.c         | 886 +++++++++++++++++++++++++
+ drivers/soc/pensando/kpcimgr_api.h     | 172 +++++
+ drivers/soc/pensando/kpcimgr_sysfs.c   | 277 ++++++++
+ 6 files changed, 1389 insertions(+)
+ create mode 100644 drivers/soc/pensando/kpci_constants.h
+ create mode 100644 drivers/soc/pensando/kpcimgr.c
+ create mode 100644 drivers/soc/pensando/kpcimgr_api.h
+ create mode 100644 drivers/soc/pensando/kpcimgr_sysfs.c
+
+diff --git a/arch/arm64/boot/dts/pensando/elba.dtsi b/arch/arm64/boot/dts/pensando/elba.dtsi
+index c05151670e1f..2c065d046f27 100644
+--- a/arch/arm64/boot/dts/pensando/elba.dtsi
++++ b/arch/arm64/boot/dts/pensando/elba.dtsi
+@@ -206,6 +206,17 @@
+ 			};
+ 		};
+ 
++		kpcimgr@20000000 {
++			compatible = "pensando,kpcimgr";
++			msi-parent = <&its 0xf>;
++			shmem-index = <3>;
++			hwmem-index = <2>;
++			reg =  <0x0 0x20000000 0x0 0x00380000   /* PXB Base */
```

Comment:
Don't change it but one of many changes for upstream was no leading zeros which means 0x00380000 => 0x380000. Somewhere down the road when we base off of linux 6.x all their personal preferences will be pulled in.